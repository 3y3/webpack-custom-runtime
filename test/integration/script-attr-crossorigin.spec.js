'use strict';

const path = require('path');
const diff = require('diff');
const webpack = require('./utils/webpack');
const WebpackCustomRuntime = require('../../').CustomRuntimePlugin;

const mainOutputDir = path.join(__dirname, 'fixtures', 'dist');

describe('ScriptAttr(CrossOrigin)', function() {

    function cit(name, options, config) {
        test(name, async function() {
            const base = webpack({
                plugins: [
                    new WebpackCustomRuntime({ behavior: '' }),
                ]
            });
            const curr = webpack(Object.assign({
                plugins: [
                    new WebpackCustomRuntime({ behavior: '' }),
                    new WebpackCustomRuntime.ScriptAttr.CrossOrigin(options),
                ]
            }, config));

            await Promise.all([ base.result, curr.result ]);

            const baseDir = base.fs.readdirSync(mainOutputDir);
            const currDir = curr.fs.readdirSync(mainOutputDir);

            expect(baseDir).toEqual(currDir);

            currDir.forEach((file) => {
                const linesDiff = diff.createTwoFilesPatch(
                    file, file,
                    base.fs.readFileSync(mainOutputDir + '/' + file).toString(),
                    curr.fs.readFileSync(mainOutputDir + '/' + file).toString(),
                    '', '',
                    { context: 0 }
                );

                expect(linesDiff).toMatchSnapshot(file);
            });
        });
    }

    cit('should not to override script crossorigin by default', {});

    cit('should override script crossorigin from options', { value: 'with-credentials' });

    cit('should override script crossorigin from config', {}, {
        output: {
            path: mainOutputDir,
            filename: '[name].js',
            crossOriginLoading: 'anonymous'
        }
    });
});
