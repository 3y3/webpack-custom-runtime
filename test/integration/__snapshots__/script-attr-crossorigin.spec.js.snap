// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptAttr(CrossOrigin) should not to override script crossorigin by default: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should not to override script crossorigin by default: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should not to override script crossorigin by default: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should not to override script crossorigin by default: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should not to override script crossorigin by default: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from config: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from config: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from config: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from config: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from config: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ -78,0 +78,4 @@
+/******/ 	  if (url.indexOf(window.location.origin + '/') !== 0) {
+/******/ 	    result.crossOrigin = 'anonymous';
+/******/ 	  }
+/******/
@@ -220,1 +224,3 @@
-/******/
+/******/ 				if (script.src.indexOf(window.location.origin + '/') !== 0) {
+/******/ 					script.crossOrigin = \\"anonymous\\";
+/******/ 				}
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from options: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from options: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from options: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from options: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(CrossOrigin) should override script crossorigin from options: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ -78,0 +78,4 @@
+/******/ 	  if (url.indexOf(window.location.origin + '/') !== 0) {
+/******/ 	    result.crossOrigin = 'with-credentials';
+/******/ 	  }
+/******/
"
`;
